src/opera/pge/dist_s1/dist_s1_pge.py

```diff
@@ -55,19 +69,90 @@ class DistS1PostProcessorMixin(PostProcessorMixin):
+                              r'(?P<acq_day>\d{2})T(?P<acq_hour>\d{2})(?P<acq_minute>\d{2})(?P<acq_second>\d{2})Z)_'
+                              r'(?P<creation_ts>(?P<cre_year>\d{4})(?P<cre_month>\d{2})(?P<cre_day>\d{2})T'
+                              r'(?P<cre_hour>\d{2})(?P<cre_minute>\d{2})(?P<cre_second>\d{2})Z)_(?P<sensor>S1)_'
+                              r'(?P<spacing>30)_(?P<product_version>v\d+[.]\d+[.]\d+))')
+
+        granule_filename_pattern = (product_id_pattern + rf'((_(?P<layer_name>{"|".join(self._output_layer_names)}))|'
+                                                         r'_BROWSE)?[.](?P<ext>tif|tiff|png)$')
+
+        product_id = re.compile(product_id_pattern + r'$')
+        granule_filename = re.compile(granule_filename_pattern)
+
+        oputput_product_path = abspath(self.runconfig.output_product_path)
+        output_products = []
+
+        for file in os.listdir(oputput_product_path):
+            dir_path = join(oputput_product_path, file)
+            if isdir(dir_path) and product_id.match(file) is not None:
+                bands = []
+                generated_band_names = []
+
+                for granule in os.listdir(dir_path):
+                    granule_path = join(dir_path, granule)
+                    if isfile(granule_path):
+                        match_result = granule_filename.match(granule)
+
+                        if match_result is None:  # or match_result.groupdict()['ext'] != 'tif':
+                            error_msg = f'Invalid product filename {granule}'
+                            self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
+                        else:
+                            if os.stat(granule_path).st_size == 0:
+                                error_msg = f'Output file {granule_path} is empty.'
+                                self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
+                            elif match_result.groupdict()['layer_name'] not in self._output_layer_names:
+                                error_msg = f'Invalid layer name "{match_result.groupdict()["layer_name"]}" in output.'
+                                self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
+                            else:
+                                bands.append(granule_path)
+                                generated_band_names.append(match_result.groupdict()['layer_name'])
+
+                if len(bands) != len(self._output_layer_names):
+                    error_msg = f'Incorrect number of output bands generated: {len(generated_band_names)}'
+                    self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
+                else:
+                    output_products.append(dir_path)
+
+        if len(output_products) != 1:
+            error_msg = f'Incorrect number of output products generated: {len(output_products)}'
```

Comment:
For clarity, let's use the term "output granules" here since we're talking about directories rather than products (the individual files _within_ a granule).

Reply:
Is that for the error message or should that be reflected in variable names too?